net/tsdial/tsdial.go

```diff
@@ -91,10 +113,51 @@ func (d *Dialer) SetExitDNSDoH(doh string) {
 	}
 }
 
+func (d *Dialer) Close() {
```

Comment:
Close without a return isn't an `io.Closer` which is sometimes annoying

Reply:
This is where I actually wanted to add the error return :)